### PR TITLE
Added docs for numbers in jsescOption in babel-generator

### PR DESCRIPTION
**Pull request by sidntrivedi012**
Docs related to https://github.com/babel/babel/pull/11028 .

/cc: @nicolo-ribaudo @JLHwung PTAL. Thanks.

### CODE REVIEW

File: docs/generator.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ minified               | boolean  | `false`         | Should the output be minif
 concise                | boolean  | `false`         | Set to `true` to reduce whitespace (but not as much as `opts.compact`)
 filename               | string   |                 | Used in warning messages
 jsonCompatibleStrings  | boolean  | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";
+numbers                | string   | `decimal`       | Set `numbers: octal` in `jsescOption` to get the octal integer representation of any numeric value. Other supported options are `decimal, binary` and `hexadecimal`.
```
Comment (JLHwung) [inline docs/generator.md]: I think we should document `jsescOption` instead as this table is for the top level generator options. We can link to https://github.com/mathiasbynens/jsesc#api for the shape of `jsescOption`.

Comment (sidntrivedi012) [reply docs/generator.md]: Made the changes. PTAL.

File: docs/generator.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ minified               | boolean  | `false`         | Should the output be minif
 concise                | boolean  | `false`         | Set to `true` to reduce whitespace (but not as much as `opts.compact`)
 filename               | string   |                 | Used in warning messages
 jsonCompatibleStrings  | boolean  | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";
+jsescOption            | string   |                 | Given some data, jsesc returns an ASCII-safe stringified representation of that data. Refer https://github.com/mathiasbynens/jsesc#api for the shape of `jsescOption` and more details.
```
Comment (existentialism) [inline docs/generator.md]: I think it's important to describe `jsescOption` within the context of @babel/generator, maybe something like:
 
```suggestion
jsescOption            | string   |                 | We leverage `jsesc` for getting ASCII-safe output when stringif-ying values. You can customize `jsesc` by [passing options](https://github.com/mathiasbynens/jsesc#api) to it.
```

?

Comment (sidntrivedi012) [reply docs/generator.md]: @existentialism Made the changes as suggested. PTAL. Thanks :)

File: docs/generator.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ minified               | boolean  | `false`         | Should the output be minif
 concise                | boolean  | `false`         | Set to `true` to reduce whitespace (but not as much as `opts.compact`)
 filename               | string   |                 | Used in warning messages
 jsonCompatibleStrings  | boolean  | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";
-numbers                | string   | `decimal`       | Set `numbers: octal` in `jsescOption` to get the octal integer representation of any numeric value. Other supported options are `decimal, binary` and `hexadecimal`.
+jsescOption            | string   |                 | Given some data, jsesc returns an ASCII-safe stringified representation of that data. Refer https://github.com/mathiasbynens/jsesc#api for the shape of `jsesOption` and more details.
```
Comment (JLHwung) [inline docs/generator.md]:
```suggestion
jsescOption            | object   |                 | Use `jsesc` to process literals. Refer https://github.com/mathiasbynens/jsesc#api for the shape of `jsesOption` and more details.
```

Comment (existentialism) [reply docs/generator.md]: @JLHwung seems like your comment is out of date?

Also, your suggestion could be misread as "_if_ you pass an object, we'll use `jsesc` to process literals".

@sidntrivedi012 he's right re: string -> object though, we should make that change.

Comment (JLHwung) [reply docs/generator.md]: @existentialism We use `jsesc` unconditionally on string literals if `jsescOption` presents, unlike numeric literals which is applied only when `jsesecOption.number` presents.

Comment (existentialism) [reply docs/generator.md]: > Use `jsesc` to process literals.

I was merely pointing out that the above statement could be read as if we only do it if you pass options, which isn't true?

Comment (sidntrivedi012) [reply docs/generator.md]: @existentialism @JLHwung Sorry for the delay in making changes. PTAL. Thanks :)

Comment (sidntrivedi012) [reply docs/generator.md]: @existentialism @JLHwung Sorry for the delay in making changes. PTAL. Thanks :)

File: website/versioned_docs/version-7.0.0/generator.md [hunk 4]
```diff
@@ -40,6 +40,7 @@ minified               | boolean  | `false`         | Should the output be minif
 concise                | boolean  | `false`         | Set to `true` to reduce whitespace (but not as much as `opts.compact`)
 filename               | string   |                 | Used in warning messages
 jsonCompatibleStrings  | boolean  | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";
+jsescOption            | object   |                 | Use `jsesc` to process literals. `jsesc` is applied to numbers only if `jsescOption.numbers` is present. You can customize `jsesc` by [passing options](https://github.com/mathiasbynens/jsesc#api) to it.
```
Comment (nicolo-ribaudo) [inline website/versioned_docs/version-7.0.0/generator.md]: (This is for versions _before_ 7.9.0)

```suggestion
jsescOption            | object   |                 | Use `jsesc` to process string literals. You can customize `jsesc` by [passing options](https://github.com/mathiasbynens/jsesc#api) to it.
```

File: website/versioned_docs/version-6.26.3/generator.md [hunk 5]
```diff
@@ -42,6 +42,7 @@ quotes                 | `'single'` or `'double'` | autodetect based on `ast.tok
 filename               | string   |                 | Used in warning messages
 flowCommaSeparator     | boolean  | `false`         | Set to `true` to use commas instead of semicolons as Flow property separators
 jsonCompatibleStrings  | boolean  | `false`         | Set to true to run `jsesc` with "json": true to print "\u00A9" vs. "©";
+jsescOption            | object   |                 | Use `jsesc` to process literals. `jsesc` is applied to numbers only if `jsescOption.numbers` is present. You can customize `jsesc` by [passing options](https://github.com/mathiasbynens/jsesc#api) to it.
```
Comment (nicolo-ribaudo) [inline website/versioned_docs/version-6.26.3/generator.md]: This isn't supported in Babel 6.

**Comment by sidntrivedi012** [comment]
@nicolo-ribaudo PTAL now.

**Pull request merged**